### PR TITLE
feat(integrations): VSTS Issue Sync (pre-existing functionality)

### PR DESCRIPTION
**Pull request by lauryndbrown**
Added the issue sync functionality from the Vsts plugin in [sentry_plugins](https://github.com/getsentry/sentry-plugins/blob/master/src/sentry_plugins/vsts/plugin.py) to the VSTS integration. This PR is mostly porting pre-existing functionality to the new integration.

### CODE REVIEW

File: src/sentry/integrations/vsts/integration.py [hunk 1]
```diff
@@ -121,13 +131,18 @@ def build_integration(self, state):
 
         scopes = sorted(VSTSIdentityProvider.oauth_scopes)
         return {
-            'name': project['name'],
-            'external_id': project['id'],
+            'name': account['AccountName'],
+            'external_id': account['AccountId'],
             'metadata': {
-                'domain_name': instance,
+                'domain_name': instance,  # This has become redundant
```
Comment (lauryndbrown) [inline src/sentry/integrations/vsts/integration.py]: Not redunant need to remove this.

File: src/sentry/integrations/vsts/integration.py [hunk 2]
```diff
@@ -121,13 +135,18 @@ def build_integration(self, state):
...
         scopes = sorted(VSTSIdentityProvider.oauth_scopes)
         return {
-            'name': project['name'],
-            'external_id': project['id'],
+            'name': account['AccountName'],
+            'external_id': account['AccountId'],
             'metadata': {
                 'domain_name': instance,
                 'scopes': scopes,
+                'default_project': {
```
Comment (lauryndbrown) [inline src/sentry/integrations/vsts/integration.py]: Will just store the id for the default_project and request info for name as needed

**Comment by lauryndbrown** [comment]
This is what the create issue looks like. not sure why the default value looks grey, but that's the name of an actual project that has been populated. 

screen shot 2018-06-15 at 3 12 29 pm

File: src/sentry/integrations/vsts/issues.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        # link = absolute_uri(group.get_absolute_url())
+        try:
+            created_item = client.create_work_item(
+                instance=self.instance,
+                project=project,
+                title=title,
+                # Decriptions cannot easily be seen. So, a comment will be added as well.
+                description=markdown(description),
+                comment=markdown(description)
+                # link=link,
```
Comment (lauryndbrown) [inline src/sentry/integrations/vsts/issues.py]: @macqueen did we still want linking? if not I'll pull it out

Comment (macqueen) [reply src/sentry/integrations/vsts/issues.py]: uhhh i don't know what it is, so i think just do whatever the old integration did! 😬

Comment (lauryndbrown) [reply src/sentry/integrations/vsts/issues.py]: I'll just leave the comments then lol

File: requirements-base.txt [hunk 4]
```diff
@@ -67,3 +67,5 @@ redis-py-cluster>=1.3.4,<1.4.0
 jsonschema==2.6.0
 # for bitbucket client
 unidiff>=0.5.4
+# for vsts repo
+mistune
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply requirements-base.txt]: That was the requirement from sentry plugins. I'll look through the history and see if there's any reason why.

[the remark this answers is not in the document]
Comment (mattrobenolt) [reply requirements-base.txt]: Probably becuase nobody said anything. :)

We pin to `mistune==0.7.2` in our production, so you can do like, `mistune>0.7,<0.8`.

File: requirements-base.txt [hunk 5]
```diff
@@ -67,3 +67,5 @@ redis-py-cluster>=1.3.4,<1.4.0
 jsonschema==2.6.0
 # for bitbucket client
 unidiff>=0.5.4
+# for vsts repo
+mistune>0.7,<0.9
```
Comment (lauryndbrown) [inline requirements-base.txt]: @mattrobenolt We only use mistune in VSTS to handle markdown of a Vsts description. Looking at the change log, we don't seem to use any features that have been affected. I see no reason not to include the most recent version.

Comment (mattrobenolt) [reply requirements-base.txt]: Cool, thanks for looking into this. We should follow up after this is merged on our internal `getsentry` version which is pinned at `0.7.2` and get it to `0.8.3`.

**Review by mattrobenolt — APPROVED** [review]
Seems ok to me. I dunno anything about VSTS.

🍕

**Pull request merged**